src/image_gen/imagegenutils.cpp

```diff
@@ -68,68 +68,130 @@ std::variant<absl::Status, std::optional<resolution_t>> getDimensions(const std:
     }
     return std::make_pair(leftInt.value(), rightInt.value());
 }
-std::variant<absl::Status, std::optional<resolution_t>> getDimensions(const ovms::HttpPayload& payload) {
-    auto sizeIt = payload.parsedJson->FindMember("size");
-    if (sizeIt == payload.parsedJson->MemberEnd()) {
+std::variant<absl::Status, std::optional<resolution_t>> getDimensions(const rapidjson::Document& payload) {
```

Comment:
Maybe also rename argument from `payload` to `parsedJson` to stay in the context?